CMakeLists.txt

```diff
@@ -26,6 +26,15 @@ elseif(${ARCH} STREQUAL "arm")
             set(rust_target thumbv7em-zephyr-eabi)
             set(clang_target thumbv7em-unknown-none-eabi)
         endif()
+    elseif(CONFIG_CPU_CORTEX_M33)
+        if(CONFIG_FPU)
+            set(rust_target thumbv8m.main-zephyr-eabihf)
+            set(clang_target thumbv8m.main-unknown-none-eabihf)
+            set(clang_cflags "-mcpu=cortex-m33 -mthumb -mabi=aapcs -mfpu=fpv5-sp-d16 -mfloat-abi=hard")
```

Comment:
Are these cpu-specific CFLAGS necessary? Clang should be only used here for bindgen, so it should not be generating binaries with these flags.

Reply:
For me, when there is no clang_cflags, it reported they (the lib generated by zephyr and the one generated by rust) have different VFP usage.

Reply:
Can you see if https://github.com/tylerwhall/zephyr-rust/pull/49 works for you?

Using the target-json generated from Rust for the `hf` variant, I get `"features": "+fp-armv8,-fp64,-d32",` in the json, and that generates and elf with the correct vfp usage.

I'm building with `west build -p auto -b nrf5340dk_nrf5340_cpuapp samples/rust-app/` and manually enabling CONFIG_FPU in `ninja menuconfig`.

```
readelf -A ./modules/zephyr-rust/app/thumbv8m.main-zephyr-eabihf/release/librust_app.a
File: ./modules/zephyr-rust/app/thumbv8m.main-zephyr-eabihf/release/librust_app.a(compiler_builtins-190bfd2d1f14c110.compiler_builtins.5742c880-cgu.9.rcgu.o)
Attribute Section: aeabi
File Attributes
  Tag_conformance: "2.09"
  Tag_CPU_arch: v8-M.mainline
  Tag_CPU_arch_profile: Microcontroller
  Tag_ARM_ISA_use: No
  Tag_THUMB_ISA_use: Yes
  Tag_FP_arch: FPv5/FP-D16 for ARMv8
```